pkg/graph/drive.resolvers.go

```diff
@@ -501,7 +703,23 @@ func (r *queryResolver) Bucket(ctx context.Context, id string) (*db.Bucket, erro
 	var bucket db.Bucket
 	err = r.DB.NewSelect().Model(&bucket).Where("id = ?", id).Where("user_id = ?", currentUser.ID).Where("organisation_id = ?", currentUser.OrganisationID).Scan(ctx)
 	if err != nil {
-		return nil, err
+		if err == sql.ErrNoRows {
+			var share db.Share
+			err = r.DB.NewSelect().Model(&share).Where("bucket_id = ?", id).Where("shared_with = ?", currentUser.ID).Scan(ctx)
+			if err != nil {
+				return nil, err
+			}
+
+			// Fetch the bucket, if share exists
+			if &share != nil {
+				err = r.DB.NewSelect().Model(&bucket).Where("id = ?", share.BucketID).Scan(ctx)
+				if err != nil {
+					return nil, err
+				}
+			}
+		} else {
+			return nil, err
+		}
```

Comment:
This also can be cleaner. Perhaps part of the first sql query

Reply:
What does this mean?